File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/service/impl/DBMatchService.kt
package dev.tpavani.octopoints.service.impl

import dev.tpavani.octopoints.db.DbSingleton
import dev.tpavani.octopoints.db.dao.MatchDao
import dev.tpavani.octopoints.db.entity.matchEntity
import dev.tpavani.octopoints.service.DBService
import dev.tpavani.octopoints.service.model.Match
import dev.tpavani.octopoints.service.utils.GetListFilter

class DBMatchService: DBService<Match, matchEntity>() {

    override val dao: MatchDao = DbSingleton.getDb().matchDao()

    override fun entityMapper(model: Match): matchEntity {
        return matchEntity(model.getId(), model.getDescription(), model.getGameMode().ordinal, model.getPoints(), model.getSurvivors(), model.isActive())
    }

    override suspend fun getList(filter: GetListFilter): MutableList<Match> {
        if (filter.active != null){
            return dao.getMatches(filter.active!!).map { Match(it.id, it.description, Match.GameMode.values()[it.gameMode], it.points, it.survivors, it.active) }.toMutableList()
        }
        return mutableListOf()
    }
}

File: octopoints_flutter/lib/service/impl/DBMatchService.dart
import 'package:octopoints_flutter/db/DbSingleton.dart';
import 'package:octopoints_flutter/db/dao/MatchDao.dart';
import 'package:octopoints_flutter/db/entity/MatchEntity.dart';
import 'package:octopoints_flutter/service/DBService.dart';
import 'package:octopoints_flutter/service/model/Match.dart';
import 'package:octopoints_flutter/service/utils/GetListFilter.dart';

class DBMatchService extends DBService<Match, MatchEntity> {
  @override
  MatchDao get dao => DbSingleton().db.matchDao;

  @override
  MatchEntity entityMapper(Match model) {
    return MatchEntity(model.id == 0 ? null : model.id, model.description, model.gameMode.index, model.points, model.survivors, model.isActive);
  }

  @override
  Future<List<Match>> getList(GetListFilter filters) async {
    if(filters.isActive != null){
      List<MatchEntity> list = await (dao).getMatches(filters.isActive!);
      return list.map((e) => Match(e.description, GameMode.values[e.gameMode], e.points, e.survivors, isActive: e.isActive, id: e.id!)).toList();
    }
    return [];
  }

}